src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -893,7 +893,11 @@ def _prepare_deployment_properties_unmodified(cmd, deployment_scope, template_fi
         template_obj = _remove_comments_from_json(_urlretrieve(template_uri).decode('utf-8'), file_path=template_uri)
     elif template_spec:
         template_link = TemplateLink(id=template_spec, mode="Incremental")
-        template_obj = show_resource(cmd=cmd, resource_ids=[template_spec]).properties['template']
+        resource = show_resource(cmd=cmd, resource_ids=[template_spec])
```

Comment:
If this logic is only for template_spec, could we specify the `api-version` to `ResourceType.MGMT_RESOURCE_TEMPLATESPECS` here? Otherwise, it will use the latest version and may be inconsistent with `ResourceType.MGMT_RESOURCE_TEMPLATESPECS`

Reply:
Such as:
```
from azure.cli.core.profiles import get_api_version
        api_version = get_api_version(cli_ctx, ResourceType.MGMT_RESOURCE_TEMPLATESPECS)
        template_obj = show_resource(cmd=cmd, resource_ids=[template_spec], api_version=api_version).properties['template']
```